alpha/lib/model/DeliveryProfileVod.php

```diff
@@ -14,21 +14,39 @@
 	 * Functionality 
 	 * --------------------*/
 
-	protected function getPlayServerUrl()
+	protected function getPlayServerUrl($entryId)
 	{
 		return '';
 	}
 
-	protected  function generatePlayServerUrl()
+	protected  function generatePlayServerUrl($entryId)
 	{
-		$prefix = '';
+		$prefix = '/dc/'.$this->getDcId($entryId);
+
 		if($this->getDynamicAttributes()->getUiConfId())
 			$prefix .= '/uiConfId/'.$this->getDynamicAttributes()->getUiConfId();
 
 		$prefix .= '/sessionId/{sessionId}';
 
 		return $prefix;
 	}
+
+	protected function getDcId($entryId)
+	{
+		$dc = kDataCenterMgr::getCurrentDcId();
+		$flavors = assetPeer::retrieveByEntryId($entryId, array(assetType::FLAVOR), array(asset::ASSET_STATUS_READY));
+		foreach ($flavors as $flavorAsset)
+		{
+			$syncKey = $flavorAsset->getSyncKey(flavorAsset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
+			list($filesync, $local) = kFileSyncUtils::getReadyFileSyncForKey($syncKey, false, false);
+			if(!$filesync)
+			{
+				$dc = $dc ? 0 : 1;
```

Comment:
This solution is hard coded for 2 dc's,  
The proper solution should be -  1. option to give DC ID to getReadyFileSyncForKey, add function getNextDcId(id) to  kDataCenterMgr.
That way you can scan 'next dc' until one has ready file sync,